lib/hmr/lazyCompilationBackend.js

```diff
@@ -5,21 +5,37 @@
 
 "use strict";
 
-const http = require("http");
-
+/** @typedef {import("http").ServerOptions} HttpServerOptions */
+/** @typedef {import("https").ServerOptions} HttpsServerOptions */
 /** @typedef {import("../Compiler")} Compiler */
 
+/**
+ * @typedef {Object} BackendConfiguration
+ * @property {HttpServerOptions | HttpsServerOptions} serverOptions Options to be passed to the Node.js HTTP or HTTPS module.
+ * @property {number?} port Custom port for lazy compilation backend. If not defined, a random port will be used.
+ */
+
 /**
  * @param {Compiler} compiler compiler
  * @param {string} client client reference
  * @param {function(Error?, any?): void} callback callback
+ * @param {?BackendConfiguration} backendConfiguration additional options for the backend
  * @returns {void}
  */
-module.exports = (compiler, client, callback) => {
+module.exports = (compiler, client, callback, backendConfiguration) => {
 	const logger = compiler.getInfrastructureLogger("LazyCompilationBackend");
 	const activeModules = new Map();
 	const prefix = "/lazy-compilation-using-";
 
+	const isHTTPS =
+		!!backendConfiguration &&
+		!!backendConfiguration.serverOptions &&
+		typeof backendConfiguration.serverOptions["key"] !== "undefined" &&
+		typeof backendConfiguration.serverOptions["cert"] !== "undefined";
```

Comment:
All looks fine, I'm just worried about this condition, maybe there are more options here that can activate https

Reply:
You're probably correct. I was using the bare minimum from your previous comment. I can include additional type checks.

Reply:
Yep, I think it will be good

Reply:
Trying to detect HTTPS by duck typing doesn't look pretty... It might be best to split into two `backendConfiguration` parameters `httpServerOptions` and `httpsServerOptions`.

Reply:
/cc @sokra What do you think? In theory we can add `type: "http" + "https"`...

Reply:
Updated PR by splitting out the options.
I looked over the types, and there are just too many HTTPS properties to try creating special handling in the backend around it.

Reply:
@alexander-akait are you good with this change?